rmw_fastrtps_cpp/src/rmw_wait.cpp

```diff
@@ -182,15 +182,19 @@ rmw_wait(
     }
   }
 
+  // We're done with the condition variable now, but there's still a chance that something it was
+  // attached to will trigger and deadlock trying to lock the condition mutex.
+  // To avoid that we release it explicitly ASAP.
+  conditionVariable = NULL;
```

Comment:
Changing the local variable should have any effect. The condition variable is still attached to all the listeners. And locally it is not being used after this line.